### PR TITLE
bfrec: Add --policy option

### PR DESCRIPTION
**Pull request by sl-mlx**
Allow the user to choose whether bfrec updates only one or both boot
partitions.

### CODE REVIEW

File: bfinst [hunk 1]
```diff
@@ -449,7 +449,7 @@ if [ -z "${skip_boot_update}" ]; then
     # MUST be executed after 'install_grub', otherwise the newly
     # created boot option would be either cleaned up or unset
     # from default option.
-    bfrec
+    bfrec --policy dual
```
Comment (kblaiech) [inline bfinst:452]: Please notify Vlad about this new flag. He needs to add this flag in his install script

File: bfrec [hunk 2]
```diff
@@ -179,7 +191,13 @@ uefi_capsule_update()
     # order once the capsule flag is set in the 'OsIndications' variable
     # which is done by the printf command below.
     $run mkdir -p "$mount_efi/EFI/UpdateCapsule" 2>/dev/null
-    $run cp $capsule_file "$mount_efi/EFI/UpdateCapsule/$capsule_efi"
+    if [ $policy = single ]; then
```
Comment (kblaiech) [inline bfrec:194]: I beleive we can implement the same logic as mlxbf-bootctl (line 254).

$run cp $capsule_file "$mount_efi/EFI/UpdateCapsule/${capsule_efi}1"
if [ $policy = "dual" ]; then
  $run cp $capsule_file "$mount_efi/EFI/UpdateCapsule/${capsule_efi}2"
fi

What do you think?

**Review by kblaiech — COMMENTED** [review]
Looks very good to me. I'll wait to hear from you about my couple of comments and then approve the changes.

File: bfinst (hunk 1 above)
Comment (sl-mlx) [reply bfinst:452]: Done

File: bfrec (hunk 2 above)
Comment (sl-mlx) [reply bfrec]: Done

File: bfrec [hunk 3]
```diff
@@ -266,6 +285,15 @@ if [ -n "$bootctl_mode" ] || [ -n "$capsule_mode" ]; then
     fi
 fi
 
+if [ -n "$policy" ]; then
+    if [ "$policy" != single ] && [ "$policy" != dual ]; then
```
Comment (lsun100) [inline bfrec:289]: Suggest use "single" and "dual" to be consistent, Or is there any reason not to use it?

How about 
[ "$policy" != "single" -a "$policy" != "dual" ] ?
Maybe we could use the checkbashisms to see which one is better

File: bfrec [hunk 4]
```diff
@@ -179,7 +191,11 @@ uefi_capsule_update()
     # order once the capsule flag is set in the 'OsIndications' variable
     # which is done by the printf command below.
     $run mkdir -p "$mount_efi/EFI/UpdateCapsule" 2>/dev/null
-    $run cp $capsule_file "$mount_efi/EFI/UpdateCapsule/$capsule_efi"
+    $run cp $capsule_file "$mount_efi/EFI/UpdateCapsule/${capsule_efi}1"
+    if [ $policy = dual ]; then
```
[the remark this answers is not in the document]
Comment (sl-mlx) [reply bfrec:195]: Done

File: bfrec [hunk 5]
```diff
@@ -233,7 +249,10 @@ kernel_bootctl_update()
     fi
 
     $run /sbin/mlxbf-bootctl -s -d $bfb_device -b $bfb_file
-    $run /sbin/mlxbf-bootctl -s -d $bfb_device -b $bfb_file
+    if [ $policy = dual ]; then
```
[the remark this answers is not in the document]
Comment (sl-mlx) [reply bfrec:252]: Done

File: bfrec (hunk 3 above)
Comment (sl-mlx) [reply bfrec:289]: > Suggest use "single" and "dual" to be consistent, Or is there any reason not to use it?

Done.

> How about
> [ "$policy" != "single" -a "$policy" != "dual" ] ?
> Maybe we could use the checkbashisms to see which one is better

Checked both with checkbashisms, both are fine. I picked
```bash
if [ "$policy" != single ] && [ "$policy" != dual ]; then
```
to match the above line...
```bash
if [ -n "$bootctl_mode" ] || [ -n "$capsule_mode" ]; then
```

**Review by kblaiech — APPROVED** [review]

**Pull request merged**